### PR TITLE
Bootleg fix implementation for categories

### PR DESCRIPTION
**Pull request by MinnDevelopment**
- Missing docs
- Missing more research

Revisit after documentation update

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/core/entities/impl/CategoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    @Override
+    public InviteAction createInvite()
+    {
+        throw new UnsupportedOperationException("Cannot create invites for category!");
+    }
+
+    @Override
+    public RestAction<List<Invite>> getInvites()
+    {
+        return new AuditableRestAction.EmptyRestAction<>(getJDA(), Collections.emptyList());
```
Comment (Almighty-Alpaca) [inline src/main/java/net/dv8tion/jda/core/entities/impl/CategoryImpl.java]: When you can't create invites for categories, are you sure getting them works?

File: src/main/java/net/dv8tion/jda/core/events/channel/category/GenericCategoryEvent.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    public Category getCategory()
+    {
+        return category;
+    }
+
+    public String getId()
+    {
+        return Long.toUnsignedString(getIdLong());
```
Comment (Almighty-Alpaca) [inline src/main/java/net/dv8tion/jda/core/events/channel/category/GenericCategoryEvent.java]: why don't just use `catergory.getId()` ?

File: src/main/java/net/dv8tion/jda/core/entities/impl/CategoryImpl.java (hunk 1 above)
Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/core/entities/impl/CategoryImpl.java]: It is an empty rest action thus it is always an empty list.

File: src/main/java/net/dv8tion/jda/core/events/channel/category/GenericCategoryEvent.java (hunk 2 above)
Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/core/events/channel/category/GenericCategoryEvent.java]: Originally I only had the ids but then i realized that all events have a Category object anyway.

File: src/main/java/net/dv8tion/jda/core/entities/impl/CategoryImpl.java (hunk 1 above)
Comment (Almighty-Alpaca) [reply src/main/java/net/dv8tion/jda/core/entities/impl/CategoryImpl.java]: right...

**Pull request merged**